tpu_inference/layers/jax/moe/moe.py

```diff
@@ -46,14 +62,17 @@ def __call__(self, x_TD: Float):
         router_act = modeling_flax_utils.ACT2FN[self.router_act]
         router_logits_TE = jnp.einsum('TD,DE -> TE', x_TD,
                                       self.kernel_DE.value)
-        weights_TX, selected_experts_TX = jax.lax.top_k(
-            router_logits_TE, self.num_experts_per_tok)
-        if self.router_act != "sigmoid":  # sigmoid does not accept axis argument.
-            normalized_weights_TX = router_act(weights_TX.astype(self.dtype),
-                                               axis=-1)
+        if self.use_moe_kernel:
+            return router_logits_TE
         else:
-            normalized_weights_TX = router_act(weights_TX.astype(self.dtype))
-        return normalized_weights_TX, selected_experts_TX
+            weights_TX, selected_experts_TX = jax.lax.top_k(
```

Comment:
even when `self.use_moe_kernel == False`, recommend changing the code so that `Router` will always only return `router_logits_TE` and let the subsequent layers figure out how to deal with other parts - such as topk.